esp-hal-smartled/Cargo.toml

```diff
@@ -14,18 +14,18 @@ targets  = ["riscv32imac-unknown-none-elf"]
 [dependencies]
 defmt             = { version = "0.3.8", optional = true }
 document-features = "0.2.10"
-esp-hal           = "0.20.0"
+esp-hal           = ">=0.20.0, <=0.21.0"
```

Comment:
Any reason why to exclude esp-hal 0.21.1?

Reply:
hmm, no. I just wonder whether <=0.21 or <=0.21.* would be cleaner/better (or even does the job)

Reply:
https://docs.rs/semver/latest/semver/enum.Op.html The first way works :)

Reply:
Heh I see! It seems a bit misleading before a good Samaritan points you to the SemVer docs 😉

Reply:
Oh it seems something is missing, this bit me:
```
error: failed to select a version for `esp-hal`.
    ... required by package `esp-hal-smartled v0.13.0`
    ... which satisfies dependency `esp-hal-smartled = "^0.13.0"` of package `test-xtensa-nostd v0.1.0 (/home/being/code/practice/rust/test-xtensa-nostd)`
versions that meet the requirements `^0.20.0` are: 0.20.1

the package `esp-hal` links to the native library `esp-hal`, but it conflicts with a previous package which links to `esp-hal` as well:
package `esp-hal v0.21.1`
    ... which satisfies dependency `esp-hal = "^0.21.1"` of package 
```

Reply:
I just fixed the dependency specifier to also allow higher versions.. Can you try 0.14? I also bumped the minor version of the crate itself.

Reply:
It worked fine! I had to set in `Cargo.toml`:
```
esp-hal-smartled = { git = "https://github.com/robamu/esp-hal-community.git", rev = "3562306", features = ["esp32s3"] }
```